pkg/reconciler/managed/api_test.go

```diff
@@ -364,26 +366,37 @@ func TestRetryingCriticalAnnotationUpdater(t *testing.T) {
 		o   client.Object
 	}
 
+	setAnnotation := func(obj client.Object) error {
+		obj.SetAnnotations(map[string]string{"getcalled": "true"})
+		return nil
+	}
+
 	cases := map[string]struct {
-		reason string
-		c      client.Client
-		args   args
-		want   error
+		reason        string
+		c             *test.MockClient
+		args          args
+		want          error
+		wantGetCalled bool
```

Comment:
Nit: Typically once we want more than one thing we nest them into a `want` struct (similar to `args`).

Reply:
fixed